deliver/lib/deliver/submit_for_review.rb

```diff
@@ -42,7 +42,8 @@ def submit!(options)
         end
       else
         UI.message("Selecting the latest build...")
-        build = wait_for_build(app, app_version)
+        target_app_version = options[:use_latest_build] ? nil : app_version
+        build = wait_for_build(app, target_app_version)
```

Comment:
This should probably be moved into https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/detect_values.rb#L63-L67

Reply:
OK, moved the logic as you pointed out.☺️